app/code/Academy/SmsSubscription/Controller/Manage/Add.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Academy\SmsSubscription\Controller\Manage;
+
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\ActionInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Controller\Result\RedirectFactory;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\State\UserLockedException;
+use Magento\Framework\Validator\Exception as ValidatorException;
+
+
+class Add extends \Magento\Framework\App\Action\Action implements ActionInterface
+{
+
+    protected $_pageFactory;
+    protected $_collection;
+    protected $_smsSubscriptionFactory;
+    protected $_smsSubscriptionResource;
+    private $_redirectFactory;
+    protected Session $customerSession;
+
+
+    public function __construct(
+        Session $customerSession,
+        Context $context,
+        \Magento\Framework\View\Result\PageFactory $pageFactory,
```

Comment:
Zachęcam do używania `use` tak aby typowanie było bardziej przejrzyste (nie jest to oczywiście obowiązkowe, to tylko sugestia).